web3_utils/retryable_eth_module.py

```diff
@@ -50,6 +51,20 @@ def before(retry_state: "RetryCallState"):
     retry_state.start_time = time.monotonic()
 
 
+def is_retryable_http_error(e) -> bool:
+    """Check for retryable HTTP status codes"""
+    return isinstance(e, HTTPError) and e.response.status_code in [
+        HTTPStatus.TOO_MANY_REQUESTS,
+        HTTPStatus.BAD_GATEWAY,
+        HTTPStatus.GATEWAY_TIMEOUT,
```

Comment:
Suggestion from an AI bot, these HTTP status codes can be retried:
```
    408 Request Timeout
    425 Too Early
    429 Too Many Requests
    500 Internal Server Error
    502 Bad Gateway
    503 Service Unavailable
    504 Gateway Timeout
```

I am skeptical about 500 though, since 500 can be thrown for various reasons due to sloppy engineering.